datafusion/physical-plan/src/lib.rs

```diff
@@ -116,12 +116,7 @@ pub mod udaf {
 /// [`required_input_ordering`]: ExecutionPlan::required_input_ordering
 pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
     /// Short name for the ExecutionPlan, such as 'ParquetExec'.
-    fn name(&self) -> &'static str
-    where
-        Self: Sized,
-    {
-        Self::static_name()
-    }
+    fn name(&self) -> &'static str;
```

Comment:
If we are going to change this signature, perhaps we can change it to return a `&str` so anyone who implements ExeutionPlan can return whatever string they want. 

Like
```suggestion
    fn name(&self) -> &str;
```